### PR TITLE
Mitch/product self hosted links

### PR DESCRIPTION
**Pull request by MitchellGerdisch**
## Checklist:

- [X] I have manually confirmed that all new links work.
- [X] I have reviewed the [style guide](../STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](../BLOGGING.md).

## Fixes
Fixes an incorrect link on https://www.pulumi.com/product/self-hosted/ for K8s and adds in a missing link for GKE as well as a link to the main self-hosted page in guides.

### CODE REVIEW

**Comment by MitchellGerdisch** [comment]
Note: The link to the updated page is here:
http://pulumi-hugo-origin-pr-2671-bcd58914.s3-website.us-west-2.amazonaws.com/product/self-hosted/

File: themes/default/layouts/product/self-hosted.html [hunk 1]
```diff
@@ -55,15 +55,15 @@ <h5 class="my-6">Docker</h5>
                         <img class="h-12" src="/logos/tech/gcp-logo.svg" alt="Google Cloud" />
                     </div>
                     <h5 class="my-6">Google Cloud</h5>
-                    <a href="{{ relref $pageContext "/contact" }}" class="btn-secondary">Contact Us</a>
+                    <a href="{{ relref $pageContext "/docs/guides/self-hosted/gke-hosted" }}" class="btn-secondary">Google Docs</a>
```
Comment (susanev) [inline themes/default/layouts/product/self-hosted.html:58]: this does not appear that we are sending them to google docs, am i missing something?

Comment (susanev) [reply themes/default/layouts/product/self-hosted.html:58]: ohh lol, i think we might want to make his link text more descriptive
maybe... Google Cloud Self-hosted docs?

**Review by susanev — APPROVED** [review]

File: themes/default/layouts/product/self-hosted.html (hunk 1 above)
Comment (MitchellGerdisch) [reply themes/default/layouts/product/self-hosted.html:58]: I was following the naming used for the other buttons. 
So, I would want to change all the other buttons as well.
That said, I think anything more than 2 words may make for an unruly, large button.

**Review by ericrudder — APPROVED** [review]
:)

File: themes/default/layouts/product/self-hosted.html (hunk 1 above)
Comment (susanev) [reply themes/default/layouts/product/self-hosted.html:58]: its just that "Google Docs" is a product...

**Pull request merged**